### PR TITLE
feat(Timeout): adding timeout option to gateway calls

### PR DESCRIPTION
**Pull request by eturino**
also refactoring to use keyword arguments

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -268,6 +268,26 @@ end
...
+```ruby
+  gateway.perform :get, path: '/this/is/required' body: nil, json_body: true, timeout: 10 
+```
+
+The internals are adapted but if a client app was mocking Gateway's `perform` directly, this could be a breaking change.
+- added the `timeout` option to perform gateway (and DAO methods). It will add timeout to the Faraday request
+
+```ruby
+def connect(method, path:, body: nil, json_body: true, timeout: nil)
+  # binding.pry if method == :delete
```
Comment (kubum) [inline README.md]: binding.pry?

File: lib/artirix_data_models/gateways/data_gateway.rb [hunk 2]
```diff
@@ -39,30 +39,33 @@ def call(method, path, json_body: true, response_adaptor: nil, body: nil, fake:
...
-  def perform(method, path, body = nil, json_body = true)
-    response = connect(method, path, body, json_body)
+  def perform(method, path:, body: nil, json_body: true, timeout: nil)
+    response = connect(method, path: path, body: body, json_body: json_body, timeout: timeout)
     treat_response(response, method, path)
   end
 
-  def connect(method, path, body = nil, json_body = true)
+  def connect(method, path:, body: nil, json_body: true, timeout: nil)
     # binding.pry if method == :delete
```
Comment (kubum) [inline lib/artirix_data_models/gateways/data_gateway.rb]: commented code?

File: lib/artirix_data_models/gateways/data_gateway.rb [hunk 3]
```diff
@@ -39,30 +39,33 @@ def call(method, path, json_body: true, response_adaptor: nil, body: nil, fake:
...
     connection.send(method, path) do |req|
+
+      req.options.timeout = timeout if timeout.present?
+
       # binding.pry if method == :delete
```
Comment (kubum) [inline lib/artirix_data_models/gateways/data_gateway.rb]: also commented code?

File: lib/artirix_data_models/gateways/data_gateway.rb (hunk 2 above)
Comment (eturino) [inline lib/artirix_data_models/gateways/data_gateway.rb]: woops

File: lib/artirix_data_models/gateways/data_gateway.rb (hunk 3 above)
Comment (eturino) [inline lib/artirix_data_models/gateways/data_gateway.rb]: yeah, that was old... didn't realise it in the diff

**Pull request merged**